### ISSUE TITLE
Unable to open a socket connection with self-hosted Supabase

### ISSUE THREAD

**Issue by dhaneshsabane** [supabase-community/realtime-py#58]
# Bug report

## Describe the bug

Following the README to connect to a self-hosted Supabase fails with `TypeError: 'type' object is not subscriptable`

Error log:

```
➜ pipenv run python -m test
Traceback (most recent call last):
  File "/home/dhanesh/.pyenv/versions/3.8.12/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/dhanesh/.pyenv/versions/3.8.12/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/dhanesh/dataDrive/playground/supabase-realtime-test/test.py", line 11, in <module>
    s = Socket(URL)
  File "/home/dhanesh/dataDrive/playground/supabase-realtime-test/.venv/lib/python3.8/site-packages/realtime/connection.py", line 51, in __init__
    self.channels = cast(defaultdict[str, List[Channel]], self.channels)
TypeError: 'type' object is not subscriptable
```

## To Reproduce

1. Follow the README to create a quick script that establishes a socket connection with self-hosted Supabase
2. Run the script
3. See error

## Expected behavior

The connection should be established with Supabase.

## Screenshots

NA

## System information

- OS: Ubuntu 22.04
- Version of realtime-py: 1.0.0
- Version of Python: 3.8.12
- Version of Pipenv: 2022.6.7

## Additional context

NA

**Comment by anand2312** [comment]
This works in CPython 3.9 and above which is when [PEP 585 - Type Hinting Generics](https://peps.python.org/pep-0585/)  was implemented; but we promise support for 3.8 as well in our pyproject.toml so we have to pick one of two fixes -
1) Add a `from __future__ import annotations` statement at the top of `realtime/connection.py` as [described in the PEP](https://peps.python.org/pep-0585/#backwards-compatibility)
2) Check the running Python version using `sys.version_info` and conditionally use `typing.DefaultDict` if we're running on py < 3.9

Would you like to open a bug-fix PR @dhaneshsabane ? If not I can get around to it after sometime

**Comment by dhaneshsabane** [comment]
@anand2312 

I'll be happy to open a bug-fix PR for this. Thanks for your suggestions!

**Comment by dhaneshsabane** [comment]
@anand2312 

I implemented option 1 but that didn't work. That's when I realised that the issue is not with type hinting but with the argument that is being passed to the `cast()` method. See here - https://github.com/supabase-community/realtime-py/blob/v1.0.0/realtime/connection.py#L52

I guess the solution here is replacing `defaultdict` with `DefaultDict` from `typing`.

**Comment by anand2312** [comment]
Ah you're right the `future` import only applies to annotations, my bad. Feel free to oepn the PR whenever, I can approve and merge (sorry for the delay in getting back to you)

**Comment by anand2312** [comment]
Fixed in #60
